### ISSUE TITLE
Testing Takes too long

### ISSUE THREAD

**Issue by sid-dinesh94** [Ninjani/Data-Mining-Project#1]
Right now, testing instance by instance happens at the rate of 1 instance per second. This needs to be made faster by predicting probabilities of all instances which go to the same reducer together. @Ninjani

**Comment by Ninjani** [comment]
Is it because we're bagging 4? Test using just one classifier to predict and see how long it takes?

**Comment by sid-dinesh94** [comment]
Still takes 100 seconds for 1000 instances. Im positive that predicting all at once will reduce the time by thousands. Mostly because the number of matrix operations to be done will be cut down.

**Comment by Ninjani** [comment]
Oh oh are we loading from pickle for each and every sample?

**Comment by quizywiz** [comment]
This would def slow it down

**Comment by sid-dinesh94** [comment]
Yeah we're loading from the pickle. That's fixed if we predict together because then we only do one IO operation per classifier
